src/components/layouts/FilterBar.tsx

```diff
@@ -80,66 +79,74 @@ export const FilterBar = ({ updateViewModeSetting }: { updateViewModeSetting: (m
-              <FilterButtonGroup filterButtons={filterButtons} activeButtonIndex={activeButtonIndex} />
               <Box
                 sx={{
                   display: { xs: 'none', sm: 'block' },
                 }}
               >
-                <Selector
-                  getSelectedValue={(_newValue) => {
-                    const newValue = _newValue as IAssigneeCombined
-                    updateAssigneeValue(newValue)
-                    handleFilterOptionsChange(FilterOptions.ASSIGNEE, newValue.id as string)
-                  }}
-                  startIcon={<FilterByAsigneeIcon />}
-                  options={assignee}
-                  placeholder="Assignee"
-                  value={assigneeValue}
-                  selectorType={SelectorType.ASSIGNEE_SELECTOR}
-                  extraOption={{
-                    id: '',
-                    name: 'No assignee',
-                    value: '',
-                    extraOptionFlag: true,
-                  }}
-                  extraOptionRenderer={(setAnchorEl, anchorEl, props) => {
-                    return (
-                      <Box
-                        component="li"
-                        {...props}
-                        sx={{
-                          '&.MuiAutocomplete-option[aria-selected="true"]': {
-                            bgcolor: (theme) => theme.color.gray[100],
-                          },
-                          '&.MuiAutocomplete-option[aria-selected="true"].Mui-focused': {
-                            bgcolor: (theme) => theme.color.gray[100],
-                          },
-                        }}
-                        onClick={(e) => {
-                          updateAssigneeValue({ id: '', name: 'No assignee' })
-                          setAnchorEl(anchorEl ? null : e.currentTarget)
-                          handleFilterOptionsChange(FilterOptions.ASSIGNEE, 'none')
-                        }}
-                      >
-                        <Stack direction="row" alignItems="center" columnGap={3}>
-                          <Avatar alt="user" sx={{ width: '20px', height: '20px' }} />
-                          <Typography variant="sm" fontWeight={400}>
-                            No assignee
-                          </Typography>
-                        </Stack>
-                      </Box>
-                    )
-                  }}
-                  buttonContent={
-                    <FilterByAssigneeBtn
-                      assigneeValue={assigneeValue}
-                      updateAssigneeValue={updateAssigneeValue}
-                      handleClick={handleFilterOptionsChange}
-                    />
-                  }
-                />
+                <FilterButtonGroup filterButtons={filterButtons} activeButtonIndex={activeButtonIndex} />
               </Box>
+              {filterOptions[FilterOptions.TYPE] !== tokenPayload?.internalUserId && (
```

Comment:
This comparison looks confusing. I know we are trying to find if a specific filterOption is active or not. But why are we comparing with `tokenPayload.internalUserId`? 

```
 filterOptions: {
    [FilterOptions.ASSIGNEE]: '',
    [FilterOptions.KEYWORD]: '',
    [FilterOptions.TYPE]: '',
  },
```
Why can't we just detect if any filter is active just by checking if `filterOptions[key] !== ''`?

Reply:
We need to check specific part of FilterOptions.TYPE here. FilterOptions.TYPE contains the button group containing My tasks, My team tasks, All tasks and Client tasks button. In the above code, I applied a check if filterOptions.TYPE contains the logged in iu's id. It could have been non empty string for client's task and my team's tasks too.